http-client/src/main/java/io/airlift/http/client/jetty/HeadersListMultimap.java

```diff
@@ -0,0 +1,52 @@
+package io.airlift.http.client.jetty;
+
+import com.google.common.collect.ForwardingListMultimap;
+import com.google.common.collect.ImmutableListMultimap;
+import com.google.common.collect.ListMultimap;
+
+import javax.annotation.Nullable;
+
+import java.util.List;
+import java.util.Map;
+
+import static com.google.common.collect.Maps.uniqueIndex;
+import static java.util.Locale.ENGLISH;
+
+class HeadersListMultimap
+        extends ForwardingListMultimap<String, String>
+{
+    private final ListMultimap<String, String> delegate;
+    private final Map<String, String> keys;
+
+    public HeadersListMultimap(ListMultimap<String, String> delegate)
+    {
+        this.delegate = ImmutableListMultimap.copyOf(delegate);
+        this.keys = uniqueIndex(delegate.keySet(), value -> value.toLowerCase(ENGLISH));
+    }
+
+    @Override
+    protected ListMultimap<String, String> delegate()
+    {
+        return delegate;
+    }
+
+    @Override
+    public List<String> get(@Nullable String key)
+    {
+        return delegate.get(getKey(key));
+    }
+
+    @Override
+    public boolean containsKey(@Nullable Object key)
```

Comment:
There are only three methods that ned to be overwritten?  I would expect a bunch more.